File: Daily_Coding_Problem/Problem_1/src/Fortran/main.f90
program main
use dcp_1
implicit none
    integer, dimension(:), allocatable :: v
    integer :: n
    integer :: k
    integer :: ifile, ofile
    integer :: i
    logical :: l

    ! read input file
    open(newunit=ifile, file="input.txt", status="old", action="read")
    read(ifile, *) n
    allocate(v(n))
    read(ifile, *) (v(i), i = 1,n)
    read(ifile, *) k

    l = is_sum(v,k)

    open(newunit=ofile, file="output.txt", action="write")
    if (l) then
        write(ofile, "(a)") "True"
    else
        write(ofile, "(a)") "False"
    end if

    close(ofile)
    close(ifile)
    deallocate(v)
end program main

File: Daily_Coding_Problem/Problem_1/src/C/main.c
#include <stdlib.h>
#include <stdio.h>
#include <stdint.h>
#include <stdbool.h>

#include "dcp_1.h"

int main()
{
    uint64_t n; 
    int64_t k;
    int64_t* arr;

    FILE* ifile = fopen("input.txt", "r");
    FILE* ofile = fopen("output.txt", "w");

    int ret = fscanf(ifile, "%ld\n", &n);
    (void) ret;
    arr = (int64_t*)malloc(n * sizeof(int64_t));

    for(int i = 0; i < n; i++) {
        ret = fscanf(ifile, "\n%ld", &arr[i]);
        (void) ret;
    }

    ret = fscanf(ifile, "%ld", &k);
    (void) ret;
        
    bool res = is_sum(n, arr, k);

    if(res) {
        fprintf(ofile, "True");
    } else {
        fprintf(ofile, "False");
    }

    free(arr);
    return 0;

}
